packages/twenty-front/src/modules/object-record/hooks/useExecuteQuickActionOnOneRecord.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback } from 'react';
+import { useApolloClient } from '@apollo/client';
+import { getOperationName } from '@apollo/client/utilities';
+
+import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
+import { capitalize } from '~/utils/string/capitalize';
+
+type useExecuteQuickActionOnOneRecordProps = {
+  objectNameSingular: string;
+};
+
+export const useExecuteQuickActionOnOneRecord = <T>({
+  objectNameSingular,
+}: useExecuteQuickActionOnOneRecordProps) => {
+  const {
+    objectMetadataItem,
+    executeQuickActionOnOneRecordMutation,
+    findManyRecordsQuery,
+  } = useObjectMetadataItem({
+    objectNameSingular,
+  });
+
+  const apolloClient = useApolloClient();
+
+  const executeQuickActionOnOneRecord = useCallback(
+    async (idToExecuteQuickActionOn: string) => {
+      const executeQuickActionOndRecord = await apolloClient.mutate({
+        mutation: executeQuickActionOnOneRecordMutation,
+        variables: {
+          idToExecuteQuickActionOn,
+        },
+        refetchQueries: [getOperationName(findManyRecordsQuery) ?? ''],
```

Comment:
This one is not ideal because enrichment can occasionally after other objects (e.g. enriching a person creates a company). Should we clear everything!?